### DISCUSSION TITLE
Wide sweep of terminology changes for consistency

### DISCUSSION THREAD

**Discussion by mikhailswift** [in-toto/witness#284] [Ideas]
We have some issues with our terminology compared to the terminology the in-toto community uses. The most glaring case is "Attestation" versus "Predicate". 

What we call an "Attestation" is typically called a "Predicate" by the in-toto community. This is especially apparent in https://github.com/in-toto/attestation/issues/136 . We use the term "Attestation" in somewhat inconsistent ways internally, too. For example, we refer to the in-toto Statement as an Attestation in specific contexts, such as uploading it to Archivista or looking at it on disk. We then use the word "Attestation" to refer to specific elements in our AttestationCollection contained within the in-toto Statement.

I believe we should consider taking a sweep through all of the terminology used within Witness and become consistent with the in-toto community. Additionally, we should take some time to create a glossary of terms in our documentation.

However, changing our "AttestationCollection" to a "PredicateCollection" may cause backward compatibility issues. While Witness is still in the 0.x.y versions and no backward compatibility is guaranteed, we should still act with care.

**Comment by matglas** [comment]
I can attest to that 😉 . When people, including me would need to get a better grip on in-toto and the different parts it better to use the right terminology. It helps to be able to research the topics better and make better suggestions.
